traffic_ops/testing/api/v4/jobs_test.go

```diff
@@ -360,3 +372,843 @@ func GetTestInvalidationJobs(t *testing.T) {
+	}
+	//past start date
+	dt := time.Now()
+	dt.Format("2019-06-18 21:28:31")
+	job := testData.InvalidationJobs[0]
+	job.StartTime = &tc.Time{
+		Time:  dt.AddDate(0, 0, -1),
+		Valid: true,
+	}
+	testData.InvalidationJobs[0] = job
+	request := tc.InvalidationJobInput{
+		DeliveryService: job.DeliveryService,
+		Regex:           job.Regex,
+		StartTime:       job.StartTime,
+		TTL:             job.TTL,
+	}
+	resp, reqInf, err := TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected startTime: must be in the future - Alert %v", resp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+
+	//RFC Format past start date
+	dt = time.Now()
+	dt.Format("2019-10-12T07:20:50.52Z")
+	job = testData.InvalidationJobs[0]
+	job.StartTime = &tc.Time{
+		Time:  dt.AddDate(0, 0, -1),
+		Valid: true,
+	}
+	testData.InvalidationJobs[0] = job
+	request = tc.InvalidationJobInput{
+		DeliveryService: job.DeliveryService,
+		Regex:           job.Regex,
+		StartTime:       job.StartTime,
+		TTL:             job.TTL,
+	}
+	resp, reqInf, err = TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected startTime: must be in the future - Alert %v", resp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+
+	//Non standard Format past start date
+	dt = time.Now()
+	dt.Format("2020-03-11 14:12:20-06")
+	job = testData.InvalidationJobs[0]
+	job.StartTime = &tc.Time{
+		Time:  dt.AddDate(0, 0, -5),
+		Valid: true,
+	}
+	testData.InvalidationJobs[0] = job
+	request = tc.InvalidationJobInput{
+		DeliveryService: job.DeliveryService,
+		Regex:           job.Regex,
+		StartTime:       job.StartTime,
+		TTL:             job.TTL,
+	}
+	resp, reqInf, err = TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected startTime: must be in the future - Alert %v", resp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+
+	//unix standard format past start date
+	job = testData.InvalidationJobs[0]
+	job.StartTime = &tc.Time{
+		Time:  time.Unix(1, 0),
+		Valid: true,
+	}
+	testData.InvalidationJobs[0] = job
+	request = tc.InvalidationJobInput{
+		DeliveryService: job.DeliveryService,
+		Regex:           job.Regex,
+		StartTime:       job.StartTime,
+		TTL:             job.TTL,
+	}
+	resp, reqInf, err = TOSession.CreateInvalidationJob(request, client.RequestOptions{})
+	if err == nil {
+		t.Errorf("Expected startTime: must be in the future - Alert %v", resp.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+}
+
+func CreateTestJobsWithFutureDate(t *testing.T) {
```

Comment:
You normally check the status code response `if reqInf.StatusCode`, but this is the only function that doesn't have that check. Was there a reason for this difference?

Reply:
accidently, it was missed. I added the status code check.